deepchecks/recommender/checks/date_train_test_leakage_overlap.py

```diff
@@ -0,0 +1,171 @@
+            if dates_leaked > 0:
+                leakage_ratio = dates_leaked / test_dataset.n_samples
+                return_value = leakage_ratio
+                text = f'{format_percent(leakage_ratio)} of test data samples are in the date range ' \
+                        f'{format_datetime(min_test_date)} - {format_datetime(max_train_date)}'\
+                        f', which occurs before last training data date ({format_datetime(max_train_date)})'
+                display.append(text)
+                fig = go.Figure()
+                # Add trace for the first list
+                fig.add_trace(go.Scatter(
+                    x=sorted(train_date.tolist()),
+                    y=[1] * len(train_date.tolist()),  # Use a constant y-value for all points in list1
+                    mode='markers',
+                    name='Train timestamps',
+                    marker=dict(
+                        color='blue'
+                    )
+                ))
+
+                # Add trace for the second list
+                fig.add_trace(go.Scatter(
+                    x=sorted(test_date.tolist()),
+                    y=[2] * len(train_date.tolist()),  # Use a different constant y-value for all points in list2
+                    mode='markers',
+                    name='Validation timestamps',
+                    marker=dict(
+                        color='red'
+                    )
+                ))
+
+                # Set the layout
+                fig.update_layout(
+                    title='Temporal Comparison of train and validation timestamps.',
+                    yaxis=dict(
+                        showticklabels=False,
+                        range=[0, 3]  # Adjust the range according to the number of lists
+                    ),
+                    height=300  # Adjust the height as needed
+                )
+                # Show the plot
+                display.append(fig)
+
+                
+        else:
+            user_id =train_dataset.user_index_name
+            train_grouped = train_dataset.data.groupby(user_id)
+            test_grouped = test_dataset.data.groupby(user_id)
+
+            leakage_ratios = []
+            
+            for user_id, train_group in train_grouped:
+                if user_id in test_grouped:
+                    test_group = test_grouped.get_group(user_id)
+                    train_group.assert_datetime()
+                    max_train_date = max(train_date)
+                    min_test_date = min(test_date)
+                    dates_leaked = sum(date < max_train_date for date in test_date)
+
+                    if dates_leaked > 0:
+                        leakage_ratio = dates_leaked / len(test_group)
+                        leakage_ratios.append(leakage_ratio)
+
+                        text = f'{format_percent(leakage_ratio)} of test data samples for user {user_id} are in the date range ' \
```

Comment:
you can't really have a text line per user... this can quickly become a huge block of text.

Can you please create an appropriate display here? Perhaps we can display the distribution (even in box plot) of overlap percents? Or a distribution of the times in which the leakage is happening? I'm sure you can come up with a better idea than this.

Also, the return value should contain the overall_leakage_ratio (which should be called average_leakage_ratio), while also containing a list of user ids that contain leakage.